unit_tests/sources/declarative/incremental/test_per_partition_cursor_integration.py

```diff
@@ -872,3 +900,82 @@ def test_per_partition_cursor_within_limit(caplog):
             },
         ],
     }
+
+
+def test_parent_stream_is_updated_after_parent_record_fully_consumed():
+    source = ConcurrentDeclarativeSource(
+        source_config=ManifestBuilder()
+        .with_substream_partition_router("AnotherStream", incremental_dependency=True)
+        .with_incremental_sync(
+            "AnotherStream",
+            start_datetime="2022-01-01",
+            end_datetime="2022-02-28",
+            datetime_format="%Y-%m-%d",
+            cursor_field=CURSOR_FIELD,
+            step="P1M",
+            cursor_granularity="P1D",
+        )
+        .with_incremental_sync(
+            "Rates",
+            start_datetime="2022-01-01",
+            end_datetime="2022-02-28",
+            datetime_format="%Y-%m-%d",
+            cursor_field=CURSOR_FIELD,
+            step="P1Y",
+            cursor_granularity="P1D",
+        )
+        .with_concurrency(1)  # so that we know partition 1 gets processed before 2
+        .build(),
+        config={},
+        catalog=None,
+        state=None,
+    )
+
+    with requests_mock.Mocker() as m:
+        # Request for parent stream
+        m.get(
+            "https://api.apilayer.com/exchangerates_data/parent/latest?from=2022-01-01&to=2022-02-28",
+            json=[{"id": "1"}],
```

Comment:
Make it 2 parent records and make sure the state is only updated after the second partition

Reply:
Test updated